### ISSUE TITLE
Add Range support to file middleware

### ISSUE THREAD

**Issue by creationix** [mjackson/strata#30]
It would be great if the file middleware supported range requests.  This way strata could be used for html video streaming (with full seek support).  Here is how I implemented it most recently in vfs-http https://github.com/c9/vfs/blob/master/http-adapter/restful.js#L82-93

**Comment by mjackson** [comment]
I agree. I've wanted to put in Range support for a while now but just haven't gotten around to it yet. Thanks for the pointer to your work. Will be really helpful in implementing this.

**Comment by creationix** [comment]
I will mention that if you go for E-Tag support for the 304 implementation, then there is etag + range interaction I implemented in http-adapter/restful as well.  It took me some time reading the somewhat inconsistent specification on what a server should do in that case.
